presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/decoders/SmallintDecoder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch.decoders;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.spi.PrestoException;
+import org.elasticsearch.search.SearchHit;
+
+import java.util.function.Supplier;
+
+import static com.facebook.presto.common.type.SmallintType.SMALLINT;
+import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTICSEARCH_TYPE_MISMATCH;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SmallintDecoder
+        implements Decoder
+{
+    private final String path;
+
+    public SmallintDecoder(String path)
+    {
+        this.path = requireNonNull(path, "path is null");
+    }
+
+    @Override
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else if (value instanceof Number) {
+            long decoded = ((Number) value).longValue();
+
+            if (decoded < Short.MIN_VALUE || decoded > Short.MAX_VALUE) {
+                throw new PrestoException(ELASTICSEARCH_TYPE_MISMATCH, format("Value out of range for field '%s' of type SMALLINT: %s", path, decoded));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: decoded (declared at line 45). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 48. Consider formatting this data with &#39;%d&#39; instead.